File: src/main/java/org/dedda/bratwurst/parse/FileImportParser.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.lang.BWInstruction;
import org.dedda.bratwurst.lang.io.file.FileImport;

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
public class FileImportParser extends InstructionParser {

    @Override
    public FileImport parse(String line, int lineNumber) {
        String variableName = line.substring(2, line.length() - 2);
        FileImport fileImport = new FileImport(lineNumber, variableName);
        return fileImport;
    }
}

File: src/main/java/org/dedda/bratwurst/parse/FileImportParser.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.lang.io.file.FileImport

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
class FileImportParser : InstructionParser() {
    override fun parse(line: String, lineNumber: Int): FileImport {
        val variableName = line.substring(2, line.length - 2)
        return FileImport(lineNumber, variableName)
    }
}